tests/kubevirt_helper.go

```diff
@@ -477,24 +485,163 @@ func AreVolumeReplicasCollocated(vol *volume.Volume, globalReplicSet []*api.Repl
 	return nil
 }
 
-func CreateConfigMap() {
+// CreateConfigMap creates configmap for vm creds
+func CreateConfigMap() error {
 	// Check if a config map named kubevirt-creds exist
-	configMap, err := core.Instance().GetConfigMap("kubevirt-creds", "default")
+	configMap, err := k8sCore.GetConfigMap("kubevirt-creds", "default")
 	log.Infof("configMap: %v", configMap)
 	log.Infof("err: %v", err)
 	if err != nil {
-		configMap = &corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "kubevirt-creds",
-			},
-			Data: map[string]string{
-				"fio-vm-multi-disk": "toor",
-			},
+		if errors.IsNotFound(err) {
+			// ConfigMap does not exist, so create it
+			configMap = &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kubevirt-creds",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"fio-vm-multi-disk": "ubuntu",
+				},
+			}
+			_, err = k8sCore.CreateConfigMap(configMap)
+			if err != nil {
+				log.Infof("Failed to create config map kubevirt-creds: %v", err)
+				return err
+			}
+			log.Infof("Created config map kubevirt-creds")
+		} else {
+			// An unexpected error occurred when retrieving the ConfigMap
+			log.Infof("Failed to retrieve config map kubevirt-creds: %v", err)
+			return err
+		}
+	} else {
+		log.Infof("Config map kubevirt-creds already exists")
+	}
+	return nil
+}
+
+// WriteFilesAndStoreMD5InVM write few files in the VM and calculates the md5sum
+func WriteFilesAndStoreMD5InVM(virtualMachines []*scheduler.Context, namespace string, fileCount int, maxFileSize int) error {
+	log.Infof("Creating %d files and storing their MD5 checksums in namespace %s", fileCount, namespace)
+	createFilesCmd := fmt.Sprintf("mkdir -p ~/testfiles && cd ~/testfiles && "+
+		"rm -f ~/file_checksums.md5 && for i in $(seq 1 %d); do "+
+		"head -c $(($RANDOM %% %d)) </dev/urandom >file$i; md5sum file$i >> ~/file_checksums.md5; done", fileCount, maxFileSize)
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+		if err != nil {
+			return err
+		}
+		for _, v := range vms {
+			_, err := RunCmdInVM(v, createFilesCmd, context1.TODO())
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ValidateFileIntegrityInVM validates the md5sum of files that we wrote in the VM
+func ValidateFileIntegrityInVM(virtualMachines []*scheduler.Context, namespace string) error {
+	log.Infof("Validating file integrity in namespace %s", namespace)
+	validateFilesCmd := "cd ~/testfiles && md5sum -c ~/file_checksums.md5"
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+		if err != nil {
+			return err
+		}
+		for _, v := range vms {
+			_, err := RunCmdInVM(v, validateFilesCmd, context1.TODO())
```

Comment:
RunCmdInVM expects the VM to have a username specified in the label, but it seems that `kubevirt-fio-pvc-clone` has the username set as an annotation. Can you please verify this?

Reply:
It's there in labels as well.